python/pyspark/sql/types.py

```diff
@@ -619,7 +621,8 @@ def fromInternal(self, obj):
             # it's already converted by pickler
             return obj
         if self._needSerializeAnyField:
-            values = [f.fromInternal(v) for f, v in zip(self.fields, obj)]
+            values = [f.fromInternal(v) if n else v
```

Comment:
Can we use the similar trick on `toInternal`?

Reply:
Yea, looks we could too.

Reply:
Done.